src/hotspot/share/runtime/mutex.hpp

```diff
@@ -45,12 +45,12 @@ class Mutex : public CHeapObj<mtSynchronizer> {
   // Special low level locks are given names and ranges avoid overlap.
   enum lock_types {
        event,
-       service        = event          +   3,
+       service        = event          +   6,
        stackwatermark = service        +   3,
        tty            = stackwatermark +   3,
        special        = tty            +   3,
        oopstorage     = special        +   3,
-       leaf           = oopstorage     +   2,
+       leaf           = oopstorage     +  10,
```

Comment:
Why do we need such a big gap here? Is there any reason we can't just use the same gap between all named rankings? As it is there seems to be no rationale for the  "+ N" value used.

Reply:
The gap is pretty much arbitrary, but there's already some leaf-2 and having overlapping rankings is something we don't really want, so I changed it to 10 for now.

Reply:
Okay but is there a reason not to use the same gap?

Reply:
If the gap is 2, then leaf-2 will == oopstorage?  I could make it 3 but it doesn't actually matter.

Reply:
Obviously the gap needs to be big enough to avoid the overlap, but the presence of +1, +3, +6 and +10 just raise questions as to why different values are used. Why not make them all +10 if you need 10 slots in some areas?

Reply:
I'm going to address this with a future change because we don't have range checking for overlap yet, and they should all be some consistent and not arbitrary range.  If it was less arbitrary, I'd pick another 6 because
runtime/mutexLocker.cpp:  def(Metaspace_lock               , PaddedMutex  , leaf-3,      true,  _safepoint_check_never);
The rank 'service' had 6 because there are at least 4 locks below that.  Again, I should add range checking and maybe making them all 10 makes a lot of sense.  Locks don't nest that deeply.  I'll add this as a comment in: JDK-8176393.